src/shogun/lib/SGVector.cpp

```diff
@@ -325,6 +340,7 @@ void SGVector<T>::init_data()
 {
 	vector=NULL;
 	vlen=0;
+	gpu_vector=NULL;
 }
 
 template<class T>
```

Comment:
please make sure that you turn off `operator[]` and other methods (put a `on_gpu` check, if yes, throw error) for which things doesn't make sense when data lives on gpu.